### PR TITLE
Adding missing and fixing some other hotkeys. Should now persist the …

### PR DESCRIPTION
**Pull request by mourasman**
…Tips and Researchtips, as well

### CODE REVIEW

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/human/buildings.ts [hunk 1]
```diff
@@ -17,7 +17,8 @@ const buildings =
             Ability.Create('Train Peasant', 'btnpeasant', 'hpea', 'P', []),
             ...defaultMainBuilding(Ability.Default(), 'rhpm'),
             Ability.MultiHotkey('Upgrade to Keep/Castle', 'btncastle', 'hkee', 'U', [], ['hcas']),
-            Ability.Create('Call to Arms', 'btncalltoarms', 'amil', 'C', []),
+            Ability.UnhotkeyAbility('Call to Arms (For one or several peons)', 'btncalltoarms', 'amil', 'C', 'W'),
```
Comment (modmoto) [inline src/hot-keys/RaceSpecificHotkeys/hotkeyData/human/buildings.ts:20]: maybe do this as Multihotkey aswell. So we have only one.

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts [hunk 2]
```diff
@@ -9,7 +9,9 @@ import {
     defaultFighterAbilities,
     defaultMainBuilding, rally, shop, stop
 } from "@/hot-keys/RaceSpecificHotkeys/hotkeyData/commonHotkeys";
+
 const uproot = Ability.Create('Uproot', 'btnuproot', 'aro1', 'R', [])
+const root = Ability.Create('Root', 'btnroot', 'aroo', 'R', [])
```
Comment (modmoto) [inline src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts:14]: There is a `Unhotkey` Ability, would fit better, so we have only one key.

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts [hunk 3]
```diff
@@ -35,7 +38,8 @@ const buildings =
             ...defaultFighterAbilities,
             Ability.Create('Eat Tree', 'btneattree', 'aeat', 'E', []),
             ...Ability.Defaults(2),
-            uproot
+            uproot,
```
Comment (modmoto) [inline src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts:41]: if root is an `UnHotkey` , we do not need this here

**Review by modmoto — COMMENTED** [review]
Hey thanks a lot for the PR =D

Like I said in the review, I would not go for the tip stuff at all. We should let wc3 handle this, so we do not have any issues with languages.

File: src/hot-keys/RaceSpecificHotkeys/RaceSpecificHotkeyTab.vue [hunk 4]
```diff
@@ -203,6 +203,9 @@ export default class RaceSpecificHotkeyTab extends Vue {
           isStagedUpgrade: this.editAbility.isStagedUpgrade,
           unHotkey: this.unhotkeyPressed,
           researchHotkey: this.researchHotkeyPressed,
+          researchTip: this.editAbility.researchTip,
+          tip: this.editAbility.tip,
```
[the remark this answers is not in the document]
Comment (abeninski) [reply src/hot-keys/RaceSpecificHotkeys/RaceSpecificHotkeyTab.vue:207]: I think somebody said that there a hotkeys only for english

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts [hunk 5]
```diff
@@ -110,10 +117,11 @@ const buildings =
             Ability.Create('Research Mark of the Talon', 'btnenchantedcrows', 'reec', 'M', []),
             ...Ability.Defaults(1),
             rally,
-            ...Ability.Defaults(1),
+            Ability.Create('Hippogryph Taming', 'btntamehippogriff', 'reht', 'I', []),
```
[the remark this answers is not in the document]
Comment (mourasman) [reply src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts:120]: 🤔 maybe in the last few patches... I'm a bit out of the loop with wc3. I'll find out.

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts (hunk 2 above)
Comment (mourasman) [reply src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts:14]: ye, you're right. I must have gotten a bit confused last night. I'll fix it!

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/human/buildings.ts (hunk 1 above)
Comment (mourasman) [reply src/hot-keys/RaceSpecificHotkeys/hotkeyData/human/buildings.ts]: Well in this case, there's 2 separate Call to Arms abilities: 1 for the Town Hall and 1 for the Peasants/Militia and they're both Unhotkey's. They'll still be one button. (I had this wrong initially, I had TWO Call to Arms in the Town Hall and I just commited a fix a few minutes ago for it)

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (mourasman) [reply src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts:120]: yes, that's correct. I'll remove this! thank you! 🙏

File: src/hot-keys/RaceSpecificHotkeys/RaceSpecificHotkeyTab.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (mourasman) [reply src/hot-keys/RaceSpecificHotkeys/RaceSpecificHotkeyTab.vue:207]: What now? :D we leave it or shall I undo the tip stuff?

File: src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts (hunk 3 above)
Comment (mourasman) [reply src/hot-keys/RaceSpecificHotkeys/hotkeyData/nightelf/buildings.ts]: yep, fixed!

File: src/update-handling/FileService.ts [hunk 6]
```diff
@@ -68,29 +69,39 @@ export class FileService {
...
                     if (positions.length === 2) {
                         currentHotkey.grid = new Grid(parseInt(positions[0]), parseInt(positions[1]))
                     }
                 }
             })
 
-            if (currentHotkey.hotKey && currentHotkey.hotkeyCommand) {
-                hotkeys.push(currentHotkey)
-            }
+            hotkeys.push(currentHotkey)
```
Comment (modmoto) [inline src/update-handling/FileService.ts:104]: that will push a lot of hotkeys that are not parse correctly

File: src/update-handling/FileService.ts [hunk 7]
```diff
@@ -56,8 +56,9 @@ export class FileService {
             const content = fse.readFileSync(hotkeyFile, 'utf8').toString().split("\n");
             let currentHotkey = {} as RaceHotKey;
             content.forEach((l: string) => {
+                const lowerCaseLine = l.toLowerCase();
                 if (l.startsWith("[")) {
-                    if (currentHotkey.hotKey && currentHotkey.hotkeyCommand) {
+                    if ((currentHotkey.hotKey && currentHotkey.hotkeyCommand) || currentHotkey.researchHotkey) {
```
Comment (modmoto) [inline src/update-handling/FileService.ts:61]: dont knot about that either, is this fine?

**Comment by modmoto** [comment]
I merged the hotkey fixes and the lower fix. I think some of the chances will not do what they seem to intend. And I still dont think we should add the tips

File: src/update-handling/FileService.ts (hunk 6 above)
Comment (mourasman) [reply src/update-handling/FileService.ts:104]: 🤔 that's probably a mistake on my end. I'll double check!

File: src/update-handling/FileService.ts (hunk 7 above)
Comment (mourasman) [reply src/update-handling/FileService.ts:61]: Yes, all auras are considered "Research", for example, therefore don't have a "Hotkey" but a "Researchhotkey" instead.

**Pull request closed without merging**